src/definition/definitionProvider.ts

```diff
@@ -0,0 +1,54 @@
+import * as _ from "lodash";
+import * as path from "path";
+import * as vscode from "vscode";
+import { mavenExplorerProvider } from "../explorer/mavenExplorerProvider";
+import { MavenProject } from "../explorer/model/MavenProject";
+import { getMavenLocalRepository } from "../utils/contextUtils";
+import { ElementNode, getCurrentNode, XmlTagName } from "../utils/lexerUtils";
+
+class DefinitionProvider implements vscode.DefinitionProvider {
+  public provideDefinition(document: vscode.TextDocument, position: vscode.Position, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.Location | vscode.Location[] | vscode.LocationLink[]> {
+    const documentText: string = document.getText();
+    const cursorOffset: number = document.offsetAt(position);
+    const currentNode: ElementNode | undefined = getCurrentNode(documentText, cursorOffset);
+    if (currentNode === undefined || currentNode.contentStart === undefined) {
+      return undefined;
+    }
+
+    switch (currentNode.tag) {
+      case XmlTagName.GroupId:
+      case XmlTagName.ArtifactId:
+      case XmlTagName.Version: {
+        const siblingNodes: ElementNode[] = _.get(currentNode, "parent.children", []);
+        const artifactIdNode: ElementNode | undefined = siblingNodes.find(elem => elem.tag === XmlTagName.ArtifactId);
+        const groupIdNode: ElementNode | undefined = siblingNodes.find(elem => elem.tag === XmlTagName.GroupId);
+        const versionNode: ElementNode | undefined = siblingNodes.find(elem => elem.tag === XmlTagName.Version);
+        const groupIdHint: string | undefined = groupIdNode && groupIdNode.text;
+        const artifactIdHint: string | undefined = artifactIdNode && artifactIdNode.text;
+        const versionHint: string | undefined = versionNode && versionNode.text;
+        if (groupIdHint && artifactIdHint) {
+          let version: string | undefined;
+          const mavenProject: MavenProject | undefined = mavenExplorerProvider.getMavenProject(document.uri.fsPath);
+          if (mavenProject) {
+            version = mavenProject.getDependencyVersion(groupIdHint, artifactIdHint);
```

Comment:
`version =  mavenProject.getDependencyVersion(groupIdHint, artifactIdHint) || versionHint;`?

Reply:
what if `mavenProject` is undefined? We still need below lines.

Reply:
I mean

```
if (mavenProject) {
    version = mavenProject.getDependencyVersion(groupIdHint, artifactIdHint) || versionHint;
}

if (version) {
    const pomPath: string = localPomPath(groupIdHint, artifactIdHint, version);
    return new vscode.Location(vscode.Uri.file(pomPath), new vscode.Position(0, 0));
}
```

Reply:
In your code above, if `mavenProject` is undefined, then `version` is always undefined, we don't have a chance to assign `versionHint` to it.

Reply:
Ok, then it can be even more simpler

`version = mavenProject?.getDependencyVersion(groupIdHint, artifactIdHint) || versionHint;`

Reply:
syntactic sugar guru 😄